mirror.js

```diff
@@ -19,16 +19,19 @@ async function handleRequest(event) {
   const cacheKey = new Request(cacheUrl.toString(), request);
 
   let response = await cache.match(cacheKey);
-  const imageRequest = new Request(imageURL, {
-    headers: request.headers
-  });
-
+  
   if (!response) {
+    const imageRequest = new Request(imageURL, {
+      headers: request.headers
+    });
     response = await fetch(imageRequest, options);
     response = new Response(response.body, response);
     response.headers.append('Cache-Control', 's-maxage=604800');
     event.waitUntil(cache.put(cacheKey, response.clone()));
+  } else {
+    response = new Response(response.body, response);
   }
-
+  
+  response.headers.set('Cache-Control', 'max-age=31536000');
```

Comment:
Why 31536000?

Reply:
It is just users browser cache.. user can remove cache anytime so added one year.. as suggested in lighthouse docs:
```
The max-age directive tells the browser how long it should cache the resource in seconds. This example sets the duration to 31536000, which corresponds to 1 year: 60 seconds × 60 minutes × 24 hours × 365 days = 31536000 seconds.

When possible, cache immutable static assets for a long time, such as a year or longer.
```
https://web.dev/uses-long-cache-ttl/?utm_source=lighthouse&utm_medium=devtools

Reply:
Isn't response.headers.set going to overwrite the call with response.headers.append before?